Sources/Sentry/SentryFileManager.m

```diff
@@ -501,6 +508,16 @@ - (NSArray *)readPreviousBreadcrumbs
     return breadcrumbs;
 }
 
+#pragma mark - Contexts
+
+- (void)moveContextFileToPreviousContextFile
+{
+    SENTRY_LOG_DEBUG(@"Moving context file to previous context file");
+    @synchronized(self.contextFilePath) {
```

Comment:
`m`: We now have the logic for accessing the `contextFilePath` split in two classes. This one here uses `@synchronized`, but the `SentryScopeContextPersistentStore` doesn't use any locks. It's very easy to mess this up in the future. Anybody could write more code accessing the `contextFilePath` and `previousContextFilePath` and it's hard to realize that the `SentryScopeContextPersistentStore` also accesses this. Maybe it's better to keep the file paths only in the `SentryScopeContextPersistentStore` and only ask the `fileManager` for the base path there.